lib/explorer/polars_backend/shared.ex

```diff
@@ -133,7 +133,13 @@ defmodule Explorer.PolarsBackend.Shared do
   end
 
   def from_list(list, {:struct, fields}, name) when is_list(list) do
-    columns = Table.to_columns(list)
+    columns =
+      list
+      |> Enum.map(fn
+        nil -> Enum.map(fields, fn {k, _} -> {k, nil} end)
+        x -> x
+      end)
+      |> Table.to_columns()
```

Comment:
```suggestion
    atom_keys? =
      Enum.reduce_while(list, false, fn
        nil, acc ->
          {:cont, acc}

        [{k, _} | _], _ ->
          {:halt, is_atom(k)}
      end)

    columns =
      list
      |> Enum.map(fn
        nil ->
          if atom_keys? do
            Enum.map(fields, fn {k, _} -> {String.to_existing_atom(k), nil} end)
          else
            Enum.map(fields, fn {k, _} -> {k, nil} end)
          end

        x ->
          x
      end)
      |> Table.to_columns()
```

Reply:
@lkarthee I pushed a commit to main that simplifies this code and we don't need Table.to_columns anymore. Can you please rebase your PR? You should just need to handle nils in `cast_series` (renamed from `cast_numerics`) and add a similar clause here for handling nils:

```
        nil, columns ->
          Enum.reduce(fields, columns, fn {field, _}, columns ->
            Map.update!(columns, field, &[nil | &1])
          end)
```